modules/sfp_twilio.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:         sfp_twilio
+# Purpose:      Extract data from phone numbers.
+#
+# Author:      Krishnasis Mandal <krishnasis@hotmail.com>
+#
+# Created:     14/06/2020
+# Copyright:   (c) Steve Micallef
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+from sflib import SpiderFoot, SpiderFootPlugin, SpiderFootEvent
+import base64
+import json
+
+class sfp_twilio(SpiderFootPlugin):
+    """Twilio:Footprint,Investigate,Passive:Search Engines:apikey:Obtain data from Twilio from target phone number."""
+    
+    opts = {
+        'account_sid': '',
+        'auth_token': ''
+    }
+
+    optdescs = {
+        'account_sid': 'Twilio Account SID',
+        'auth_token': 'Twilio Auth Token'
+    }
+
+    results = None
+    errorState = False
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.results = self.tempStorage()
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+
+    def watchedEvents(self):
+        return ["PHONE_NUMBER"]
+
+    # What events this module produces
+    def producedEvents(self):
+        return ["HUMAN_NAME", "COMPANY_NAME", "RAW_RIR_DATA"]
+
+    # When querying third parties, it's best to have a dedicated function
+    # to do so and avoid putting it in handleEvent()
+    def queryPhoneNumber(self, phoneNumber):   
+        
+        token = (base64.b64encode(self.opts['account_sid'].encode('utf8') + ":".encode('utf-8') + self.opts['auth_token'].encode('utf-8'))).decode('utf-8')
+
+        headers = {
+            'Accept': "application/json",
+            'Authorization': "Basic " + token
+        }
+
+        res = self.sf.fetchUrl(
+          'https://lookups.twilio.com/v1/PhoneNumbers/' + phoneNumber,
+          headers=headers,
+          timeout=15,
+          useragent=self.opts['_useragent']
+        )
+
+        if res['code'] == '400':
+            self.sf.error("Bad request.", False)
+            return None
+        
+        if res['code'] == '429':
+            self.sf.error("API usage limit reached.", False)
+            return None
+
+        if res['code'] == '503':
+            self.sf.error("Service unavailable.", False)
+            return None
+
+        if not res['code'] == '200':
```

Comment:
If the phone number is not found, what's the return code? If 404 then this shouldn't be an error.